rtlive/sources/data_it.py

```diff
@@ -1,39 +1,210 @@
+    '7': 'Liguria',
+    '8': 'Emilia-Romagna',
+    '9': 'Toscana',
+    '10': 'Umbria',
+    '11': 'Marche',
+    '12': 'Lazio',
+    '13': 'Abruzzo',
+    '14': 'Molise',
+    '15': 'Campania',
+    '16': 'Puglia',
+    '17': 'Basilicata',
+    '18': 'Calabria',
+    '19': 'Sicilia',
+    '20': 'Sardegna',
+    '21': 'P.A. Bolzano',
+    '22': 'P.A. Trento',
+    'all': 'Italy',
+}
+
+IT_REGION_CODES = {
+    v : k
+    for k, v in IT_REGION_NAMES.items()
+}
+
+IT_REGION_POPULATION = {
+    'all': 60_244_639,
+    '1': 4_341_375,
+    '2': 125_501,
+    '3': 10_103_969,
+    '5': 4_907_704,
+    '6': 1_211_357,
+    '7': 1_543_127,
+    '8': 4_467_118,
+    '9': 3_722_729,
+    '10': 880_285,
+    '11': 1_518_400,
+    '12': 5_865_544,
+    '13': 1_305_770,
+    '14': 302_265,
+    '15': 5_785_861,
+    '16': 4_008_296,
+    '17': 556_934,
+    '18': 1_924_701,
+    '19': 4_968_410,
+    '20': 1_630_474,
+    '21': 532_080,
+    '22': 542_739,
+}
+
+LABEL_TRANSLATIONS = {
+    "curves_ylabel": "Giornalieri",
+    "testcounts_ylabel": "Tests",
+    "probability_ylabel": "Probabilità\n$R_t$>1",
+    "rt_ylabel": "$R_t$",
+    "curve_infections": "Infezioni",
+    "curve_adjusted": "Test positivi attesi",
+    "bar_positive": "Test positivi confermati",
+    "bar_actual_tests": "Dati",
+    "curve_predicted_tests": "Previsione",
+}
+IT_REGIONS = list(IT_REGION_NAMES.keys())
+
+
+def get_data_IT(run_date) -> pandas.DataFrame:
+    if run_date.date() > datetime.date.today():
+        raise ValueError("Run date is in the future. Nice try.")
+    if run_date.date() < datetime.date.today():
+        # TODO: implement downloading of historic data
+        raise NotImplementedError(
+            "Downloading with a run_date is not yet supported. "
+            f"Today: {datetime.date.today()}, run_date: {run_date}"
+        )
+    data = get_regions_data(run_date)
+    global_data = get_global_data(run_date)
+    data = data.append(global_data, ignore_index=True)
+    return data
+
+def get_global_data(run_date) -> pandas.DataFrame:
+    today_obj = datetime.strptime(run_date, '%Y-%m-%d')
+    today = today_obj.strftime('%Y%m%d')
+
+    content = requests.get(
+        "https://raw.githubusercontent.com/pcm-dpc/COVID-19/master/dati-andamento-nazionale/dpc-covid19-ita-andamento-nazionale-" + str(today) + ".csv",
+    ).content
+    today_data = pandas.read_csv(
+        io.StringIO(content.decode("utf-8")),
+        sep=",",
+        parse_dates=["data"],
+        usecols=["data", "nuovi_positivi", "tamponi"],
+    ).rename(
+        columns={
+            "data": "date",
+            "nuovi_positivi": "new_cases",
+            "tamponi": "new_tests",
+        }
+    )
+
+    yesterday_obj = today_obj - timedelta(days=1)
+    yesterday = yesterday_obj.strftime('%Y%m%d')
+    content = requests.get(
+        "https://raw.githubusercontent.com/pcm-dpc/COVID-19/master/dati-andamento-nazionale/dpc-covid19-ita-andamento-nazionale-" + str(yesterday) + ".csv",
```

Comment:
Nitpick, but this link and the one in line 101 are the same, so you could store it in a variable and just use f-strings: `requests.get(f"GLOBAL_DATA_LINK{str(yesterday)}.csv")`
Will be clearer to read and easier to maintain if the link changes 🤷‍♂️